modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java

```diff
@@ -310,7 +310,7 @@ public void visitArrayType(ArrayType arrayType) {
 
         private boolean parseAsContent(MetadataType type) {
           if (isContent) {
-            parseAttributeParameter(parameter);
+            parseObjectParameter(parameter, paramDsl);
```

Comment:
content parameters are not necesssarily objects. As a matter of fact, most of the times they're just streams. Method names should be reviewed